### PR TITLE
#142  Renew devops configs 

### LINKED ISSUE #142: Autobuild local env as on SE
В ~https://github.com/fidals/shopelectro/issues/176~ https://github.com/fidals/shopelectro/issues/269 всё реализовали.

Повтори это на Бетоне или убедись, что это реализовано

### PR DESCRIPTION
**Pull request by duker33**
Closes #142

### CODE REVIEW

File: docker/Makefile [hunk 1]
```diff
@@ -1,61 +1,103 @@
...
+# ---------------------- Deploy section ----------------------
+deploy-dev:
+	$(MAKE) create-env
+	$(MAKE) create-config
+	$(dc) pull
+	$(dc) up -d app
+	$(MAKE) build-static
+	# Create admin user with login/pass: admin/asdfjkl;
+	# Launch "collectstatic" not in static recipe because ManifestStaticStorage writes to db
+	$(dc) exec app bash -c "\
```
Comment (artemiy312) [inline docker/Makefile]: Maybe we should reuse existing receipts instead of it, what do you think?
```
$(MAKE) migrate
$(MAKE) load-fixtures
$(MAKE) collectstatic
```

File: docker/create-env.sh [hunk 2]
```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+
+# Copy the .dist environment files and wait them modifications.
```
Comment (artemiy312) [inline docker/create-env.sh]: .dist -> env_files/*.dist

File: docker/create-env.sh [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        then
+            cp $file $new_file
+        fi
+    done
+
+    if ! $QUITE
+    then
+        file_names=$(join_by , ${new_files[@]})
+        read -p \
+"Go to this files and fill them with your own values.
```
Comment (artemiy312) [inline docker/create-env.sh]: Looks too verbose. Maybe it will be better to say:
```
Configure environment files with values you need: $file_names.
Are you ready to merge them into a .env file?
```

File: docker/docker-compose.yml [hunk 4]
```diff
@@ -2,139 +2,104 @@ version: '2'
...
-  stb-selenium:
-    image: selenium/standalone-chrome-debug:3.6.0
+  selenium:
+    image: selenium/standalone-chrome-debug:3.10.0
     restart: always
     ports:
-      - '4444'
-      - '5900:5900'
+      - 4444:4444
+      - 4444
```
Comment (artemiy312) [inline docker/docker-compose.yml]: `- 4444:4444` will be enough

File: docker/env_files/app [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# It may affect on app logic. Valid values: LOCAL | CI | PROD
```
Comment (artemiy312) [inline docker/env_files/app]: It must be ignored by git

File: docker/env_files/credentials [hunk 6]
```diff
@@ -0,0 +1,28 @@
+# Docs here: https://docs.docker.com/compose/environment-variables/#/the-env-file
```
Comment (artemiy312) [inline docker/env_files/credentials]: It must be ignored by git

File: docker/env_files/paths [hunk 7]
```diff
@@ -0,0 +1,5 @@
+# Identify the dependencies folder
```
Comment (artemiy312) [inline docker/env_files/paths]: It must be ignored by git

File: docker/env_files/ports [hunk 8]
```diff
@@ -0,0 +1,6 @@
+# Exposed ports
```
Comment (artemiy312) [inline docker/env_files/ports]: It must be ignored by git

File: docker/images/python/Dockerfile.dev [hunk 9]
```diff
@@ -0,0 +1,15 @@
+FROM python:slim
+
+RUN apt-get update \
+    # wget is needed for working with ftp
```
Comment (artemiy312) [inline docker/images/python/Dockerfile.dev]: We do not work with ftp

File: docker/images/python/Dockerfile.dev [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+    # git is needed for pip
+    && apt-get install --no-install-recommends --no-install-suggests -y \
+    wget git make wkhtmltopdf xvfb gettext
```
Comment (artemiy312) [inline docker/images/python/Dockerfile.dev]: seems it will be enough`wkhtmltopdf xvfb gettext`

File: docker/images/python/Dockerfile.prod [hunk 11]
```diff
@@ -0,0 +1,8 @@
+FROM fidals/se:dev
```
Comment (artemiy312) [inline docker/images/python/Dockerfile.prod]: fidals/se:dev -> fidals/stb:dev

File: docker/Makefile [hunk 12]
```diff
@@ -1,61 +1,103 @@
...
-	$(dc) exec stb-python python manage.py migrate
+	$(dc) exec app python manage.py migrate
+
+fixtures:
+	$(dc) run --rm app bash -c "\
+		python manage.py loaddata stroyprombeton/fixtures/admin.json \
+		&& python manage.py loaddata stroyprombeton/fixtures/dump.json \
+	"
+
+prices:
```
[the remark this answers is not in the document]
Comment (duker33) [reply docker/Makefile]: @artemiy312 , let's leave `prices`, because we already have django command with the same name

File: docker/Makefile [hunk 13]
```diff
@@ -1,61 +1,103 @@
...
+build test backup restore \
+generate-production-static-data deploy
+
+
+# ---------------------- Dev section ----------------------
 migrate:
-	$(dc) exec stb-python python manage.py migrate
+	$(dc) exec app python manage.py migrate
+
+fixtures:
```
[the remark this answers is not in the document]
Comment (duker33) [reply docker/Makefile]: @artemiy312 , 
we have conflict with short vs clear commands.
I think commands should be short for the first. And clear for the second.
We type them much more often, then remember.
We all use Linux command `ls`, but not `show-files`.

New teammates will dig into Makefile anyway: with short or not commands.
So, i propose to leave short names of the commands, but add some doc for them in Makefile. We'll do it at https://github.com/fidals/shopelectro/issues/303

File: docker/Makefile [hunk 14]
```diff
@@ -1,61 +1,103 @@
...
+build-static:
+	$(dc) run --rm nodejs gulp build
+
+watch-static:
+	$(dc) run --rm nodejs gulp watch
 
-# for local env
 build: static
-	$(dc) build stb-python
+	$(dc) build app
```
[the remark this answers is not in the document]
Comment (duker33) [reply docker/Makefile]: TODO

File: docker/Makefile [hunk 15]
```diff
@@ -1,61 +1,103 @@
...
+	$(dc) up -d selenium
+	$(dc) exec app python manage.py test --liveserver=app:8001-8009 --parallel
 	$(dc) stop
 
-# for local env
 lint:
-	$(dc) up -d stb-python
-	$(d) run --rm --volumes-from stb-python --workdir=/usr/app/src -it coala/base coala
+	$(dc) up -d app
+	$(d) run --rm --volumes-from app --workdir=/usr/app/src -it coala/base coala
```
[the remark this answers is not in the document]
Comment (duker33) [reply docker/Makefile]: TODO

File: docker/Makefile (hunk 1 above)
Comment (duker33) [reply docker/Makefile]: TODO

File: docker/create-env.sh (hunk 3 above)
Comment (duker33) [reply docker/create-env.sh]: @artemiy312 , 
- `configure files with values` -> `fill files with values`
- proger: what values? Will someone give them for me? They lay in another part of system? -> `your own values`
- it's hard to parse file_names list from text, when they are on the same string

> Are you ready to merge them into a .env file?
- *bad:* proger: merge who? Files or values? ..., oh, it's the same thing
- *good:* `merge your values to .env file` -> proger: ok

> Hope you finished to configure files from list below.

this string we can remove.

So, i propose this:
```
Go to this files and fill them with your own values.
Files list: $file_names.
Ready to merge new config values to .env file? [y/n]:
```

File: docker/docker-compose.yml (hunk 4 above)
Comment (duker33) [reply docker/docker-compose.yml]: @artemiy312 , i'll leave just `- 4444` for safety

TODO

File: docker/images/python/Dockerfile.dev (hunk 9 above)
Comment (duker33) [reply docker/images/python/Dockerfile.dev]: TODO

File: docker/images/python/Dockerfile.prod (hunk 11 above)
Comment (duker33) [reply docker/images/python/Dockerfile.prod]: TODO

File: docker/images/python/Dockerfile.dev (hunk 10 above)
Comment (duker33) [reply docker/images/python/Dockerfile.dev]: `git` and `make` are required too

**Comment by duker33** [comment]
@artemiy312 , refactored drone.yml too. Pay attention to it plz

File: docker/Makefile [hunk 16]
```diff
@@ -1,61 +1,98 @@
...
 	$(dc) stop
 
-# for local env
-restore:
-	@bash ../etc/stb-backup-restore.sh
 
-# for serv env
+deploy-dev:
+	$(MAKE) create-env
+	# $(MAKE) create-config
```
Comment (artemiy312) [inline docker/Makefile]: Seems it should be uncommented

File: docker/images/python/Dockerfile.dev [hunk 17]
```diff
@@ -0,0 +1,14 @@
+FROM python:slim
+
+RUN apt-get update \
+    # git is needed for pip
+    && apt-get install --no-install-recommends --no-install-suggests -y \
+    git make wkhtmltopdf xvfb gettext
```
Comment (artemiy312) [inline docker/images/python/Dockerfile.dev]: We do not use git and make anywhere, so we should remove it.

File: docker/images/python/Dockerfile.dev [hunk 18]
```diff
@@ -0,0 +1,14 @@
...
+    && apt-get remove --purge -y git \
```
Comment (artemiy312) [inline docker/images/python/Dockerfile.dev]: This line should be removed as above

File: docker/Makefile (hunk 16 above)
Comment (duker33) [reply docker/Makefile]: it depends on the task. I'll add comment

File: docker/images/python/Dockerfile.dev (hunk 17 above)
Comment (duker33) [reply docker/images/python/Dockerfile.dev]: @artemiy312 , 

> # git is needed for pip

File: docker/images/python/Dockerfile.dev (hunk 18 above)
Comment (duker33) [reply docker/images/python/Dockerfile.dev]: Done.
